sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -726,6 +726,14 @@ private[sql] class SQLConf extends Serializable with CatalystConf with Logging {
   }
 
   /**
+   * Return the value of an optional Spark SQL configuration property for the given key. If the key
+   * is not set yet, returns None.
+   */
+  def getOptionalConf[T](entry: OptionalConfigEntry[T]): Option[T] = {
```

Comment:
Ah, that's the configuration that caused all this... so I'm wondering if that method shouldn't also return `Option[T]`. IIRC there's only one place where that method is called.